### ISSUE TITLE
panc 9.2 throws NullPointerException sometimes

### ISSUE THREAD

**Issue by loomis** [quattor/pan#6]
Reported by Gabor in SF Bug #3545788:

Below is the stack trace. The error is not deterministic, running panc again on the same sources usually succeeds. All occurances of the error I've seen complain about the same pan type (component_ofed_openib_modules).

```
java.lang.NullPointerException
[panc] at java.util.TreeMap.rotateLeft(TreeMap.java:2038)
[panc] at java.util.TreeMap.fixAfterInsertion(TreeMap.java:2105)
[panc] at java.util.TreeMap.put(TreeMap.java:559)
[panc] at org.quattor.pan.dml.data.HashResource.put(HashResource.java:103)
[panc] at org.quattor.pan.type.RecordType.setDefaults(RecordType.java:231)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.type.AliasType.setDefaults(AliasType.java:96)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.type.RecordType.setDefaults(RecordType.java:233)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.type.AliasType.setDefaults(AliasType.java:96)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.type.RecordType.setDefaults(RecordType.java:211)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.type.AliasType.setDefaults(AliasType.java:96)
[panc] at org.quattor.pan.type.FullType.setDefaults(FullType.java:153)
[panc] at org.quattor.pan.tasks.BuildTask$CallImpl.setDefaults(BuildTask.java:224)
[panc] at org.quattor.pan.tasks.BuildTask$CallImpl.call(BuildTask.java:125)
[panc] at org.quattor.pan.tasks.BuildTask$CallImpl.call(BuildTask.java:74)
[panc] at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[panc] at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[panc] at org.quattor.pan.cache.AbstractCache.retrieve(AbstractCache.java:137)
[panc] at org.quattor.pan.cache.AbstractCache.waitForResult(AbstractCache.java:171)
[panc] at org.quattor.pan.tasks.Valid1Task$CallImpl.call(Valid1Task.java:85)
[panc] at org.quattor.pan.tasks.Valid1Task$CallImpl.call(Valid1Task.java:68)
[panc] at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[panc] at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[panc] at org.quattor.pan.cache.AbstractCache.retrieve(AbstractCache.java:137)
[panc] at org.quattor.pan.cache.AbstractCache.waitForResult(AbstractCache.java:171)
[panc] at org.quattor.pan.tasks.Valid2Task$CallImpl.call(Valid2Task.java:83)
[panc] at org.quattor.pan.tasks.Valid2Task$CallImpl.call(Valid2Task.java:66)
[panc] at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[panc] at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[panc] at org.quattor.pan.cache.AbstractCache.retrieve(AbstractCache.java:137)
[panc] at org.quattor.pan.cache.AbstractCache.waitForResult(AbstractCache.java:171)
[panc] at org.quattor.pan.tasks.WriteXmlTask$CallImpl.call(WriteXmlTask.java:100)
[panc] at org.quattor.pan.tasks.WriteXmlTask$CallImpl.call(WriteXmlTask.java:69)
[panc] at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[panc] at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[panc] at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:441)
[panc] at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
[panc] at java.util.concurrent.FutureTask.run(FutureTask.java:138)
[panc] at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
[panc] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
[panc] at java.lang.Thread.run(Thread.java:662)
[panc] type lookup failed for 'component_ofed_openib_modules'
```

**Comment by gombasg** [comment]
We're still on 10.1, but no commits in the history since makes me believe 10.2 would be better.

**Comment by jrha** [comment]
This was resolved as part of #105 in 10.3.
